packages/trpc/src/procedures/forgot-password.ts

```diff
@@ -0,0 +1,139 @@
+import { TRPCError } from "@trpc/server";
+import { z } from "zod";
+
+import { hashPassword } from "@good-dog/auth/password";
+import { sendPasswordResetEmail } from "@good-dog/email/password-reset-email";
+
+import { baseProcedureBuilder } from "../internal/init";
+
+// click forgot passwd -> enter email -> receive unique link -> enter new passwd at given link
+
+const getNewPasswordResetExpirationDate = () =>
+  new Date(Date.now() + 60_000 * 15);
+
+export const sendForgotPasswordEmailProcedure = baseProcedureBuilder
+  .input(
+    z.object({
+      email: z.string().email(),
+    }),
+  )
+  .mutation(async ({ ctx, input }) => {
+    // Find user with the given email
+    const user = await ctx.prisma.user.findUnique({
+      where: {
+        email: input.email,
+      },
+    });
+
+    // If user with that email doesn't exist, throw error
+    if (!user) {
+      throw new TRPCError({
+        code: "NOT_FOUND",
+        message: `No user found with given email.`,
+      });
+    }
+
+    // Delete any existing password reset requests for the user
+    await ctx.prisma.passwordResetReq.deleteMany({
+      where: {
+        user: user,
+      },
+    });
+
+    // Create new password reset request with updated expired at time
+    const pwdResetReq = await ctx.prisma.passwordResetReq.create({
+      data: {
+        user: {
+          connect: {
+            userId: user.userId,
+          },
+        },
+        expiresAt: getNewPasswordResetExpirationDate(),
+      },
+    });
+
+    // Update the user to include the new password reset request
+    await ctx.prisma.user.update({
+      where: {
+        email: input.email,
+      },
+      data: {
+        passwordResetReq: {
+          connect: {
+            passwordResetId: pwdResetReq.passwordResetId,
+          },
+        },
+      },
+    });
+
+    // Send the password reset email
+    try {
+      await sendPasswordResetEmail(input.email, pwdResetReq.passwordResetId);
+    } catch (error) {
+      throw new TRPCError({
```

Comment:
Add the case for env.NODE_ENV = 'development' like i did with the send email procedure so we can still use this locally without a sendgrid account